pkg/pool-stable-phantom/contracts/StablePoolProtocolFees.sol

```diff
@@ -156,6 +150,52 @@ abstract contract StablePoolProtocolFees is StablePoolStorage, StablePoolRates,
             );
     }
 
+    function _getGrowthInvariants(uint256[] memory balances, uint256 lastPostJoinExitAmp)
+        internal
+        view
+        returns (
+            uint256 swapFeeGrowthInvariant,
+            uint256 totalNonExemptGrowthInvariant,
+            uint256 totalGrowthInvariant
+        )
+    {
+        // We always calculate the swap fee growth invariant, since we cannot easily know whether swap fees have
+        // accumulated or not.
+
+        swapFeeGrowthInvariant = StableMath._calculateInvariant(
+            lastPostJoinExitAmp,
+            _getAdjustedBalances(balances, true) // Adjust all balances
+        );
+
+        // For the other invariants, we can potentially skip some work. In the edge cases where none or all of the
+        // tokens are exempt from yield, there's one fewer invariant to compute.
+
+        if (_areNoTokensExempt()) {
```

Comment:
I guess _areXXX is analogous to _isXXX, but _noTokensAreExempt() and _allTokensAreExempt() sound more natural to me. (Or even drop the "are": _allTokensExempt(), _noTokensExempt().)

Reply:
That'd make them collide with the variables though :( Can I keep my names?

Reply:
Yeah; I think _areNoTokensExempt / _noTokensExempt is clearer than 
                     _noTokensAreExempt / _noTokensExempt